scripts/prepare_cop_timeseries.py

```diff
@@ -0,0 +1,215 @@
+# coding: utf-8
+r"""
+Inputs
+-------
+in_path1 : str
+    ``raw/weatherdata``: path of input directory with weather data
+in_path2 : str
+    ``raw/scalars/capacities.csv``: path of scalar data as .csv
+
+Outputs
+---------
+pandas.DataFrame
+    with timeseries of cops of air-water heat pumps
+
+Description
+-------------
+The script calculates cop timeseries of small-scale air-water heat pumps for decentralized use.
+"""
+
+import datetime
+import os
+import sys
+import itertools
+import pandas as pd
+import numpy as np
+import oemof_b3.tools.data_processing as dp
+
+# from oemof.thermal.compression_heatpumps_and_chillers import calc_cops
+
+
+def find_regional_files(path, region):
+    """
+    This function returns a list of file names in a directory that match the specified region.
+
+    Parameters
+    ----------
+    path : str
+        Path to data
+
+    region : str
+        Region (eg. Brandenburg)
+
+    Returns
+    -------
+    files_region : list
+        List of file names matching region
+    """
+    files_region = [file for file in os.listdir(path) if f"_{region}_" in file]
+    files_region = sorted(files_region)
+
+    if not files_region:
+        raise FileNotFoundError(
+            f"No data of region {region} could be found in directory: {path}."
+        )
+
+    return files_region
+
+
+def get_year(file_name):
+    """
+    This function returns a year from file name
+
+    Parameters
+    ----------
+    file_name : str
+        Name of file with year in it
+
+    Returns
+    -------
+    year : int
+        Year
+    """
+    # Add array with years to be searched for in file name
+    years_search_array = np.arange(1990, 2051)
+    newline = "\n"
+
+    year_in_file = [
+        year_searched
+        for year_searched in years_search_array
+        if str(year_searched) in file_name
+    ]
+    if len(year_in_file) == 1:
+        year = year_in_file[0]
+    else:
+        raise ValueError(
+            f"Your file {file_name} is missing a year or has multiple years "
+            f"in its name." + newline + "Please provide data for a single year "
+            "with that year in the file name."
+        )
+
+    return year
```

Comment:
I've takes this from `prepare_heat_demand.py`. Could we put it some place else?

Reply:
For the moment, let us leave it here and add a note that it is a duplicate of the functions in prepare_heat_demand. With more refactoring, this can become general enough to be moved to the dataprocessing module.

Reply:
Comment left with commit https://github.com/rl-institut/oemof-B3/pull/199/commits/b03605a6948f90d0f6568e2cdd853e9b4cfb6dee.